src/pages/signalProviders/index.js

```diff
@@ -58,9 +72,54 @@ const SignalProviders = (props) => {
     if (providerId && providerId.length === 24) {
       loadProvider();
     }
+
+    return () => {
+      setEmptySettingsAlert(false);
+    };
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [providerId]);
 
+  const loadSettings = () => {
+    if (
+      provider.id &&
+      !provider.disable &&
+      provider.exchangeInternalId === selectedExchange.internalId &&
+      quotesAvailable === "true"
+    ) {
+      const payload = {
+        token: storeSession.tradeApi.accessToken,
+        providerId: provider.id,
+        internalExchangeId: selectedExchange.internalId,
+        version: 2,
+      };
+      tradeApi
+        .providerExchangeSettingsGet(payload)
+        .then((response) => {
+          setSettings(response);
+          checkAllocated(response);
+        })
+        .catch((e) => {
+          dispatch(showErrorAlert(e));
+        });
+    }
+  };
+
+  useEffect(loadSettings, [selectedExchange.internalId, provider.id, quotesAvailable]);
+
+  /**
```

Comment:
If you could add a description for the method because I'm lost ;)

Reply:
Done.